### DISCUSSION TITLE
[SOLVED] [setupSFC] [question] [nuxt] Error with setupSFC in Nuxt 3

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [vue-macros/vue-macros#429] [Q&A] [answered]
![image](https://github.com/vue-macros/vue-macros/assets/10191085/d18eaa5a-dc70-4551-ac28-3a782ff7b9f7)

nuxt.config.ts:

```ts
import vueMacros from 'unplugin-vue-macros/vite'
import vue from '@vitejs/plugin-vue'

export default defineNuxtConfig({
  vite: {
    vue: {
      script: {
        defineModel: true,
        propsDestructure: true,
        
      },
    },
    plugins: [
      vueMacros(),
      vue({
        include: [/\.vue$/, /\.setup\.[cm]?[jt]sx?$/],
        //                   ⬆️ setupSFC pattern need to be added
      }) as any,
    ],
  },
  
   modules: [
    '@vue-macros/nuxt',
  ],
})
```

pages/index.setup.tsx:

```ts
export default () => (
  <div>
  <h1>Hello World</h1>
</div>
)
```

### Environment

OS: Windows 11 Home 10.0.22621 Build 22621
Node: 18.15.0
nuxt: 3.6.5
@vue-macros/nuxt: 1.3.19
VSCode: 1.80.1

**Comment by angelhdzmultimedia** [comment]
### Attempts to solve it

![image](https://github.com/vue-macros/vue-macros/assets/10191085/f13a584b-a1f8-4a59-beb1-73337d5e8395)

Unsuccessful.

**Comment by sxzz** [comment]
`setupSFC` is disabled by default, and you should enable it first.
See [demo](https://github.com/vue-macros/nuxt/blob/main/nuxt.config.ts)

**Comment by angelhdzmultimedia** [comment]
Sorry to bother you again. It's not working for me. Same error of the image I provided.

Tried removing `.nuxt` directory and adding the vue and vueJsx vite plugins without success.

My nuxt.config.ts:

```ts
 modules: [
    '@pinia/nuxt',
    'nuxt-quasar-ui',
    '@nuxtjs/i18n',
    '@vue-macros/nuxt',
  ],

  

  macros: {
    setupSFC: true,
  },

```

Also tried with:

```ts
 macros: {
    setupSFC: {
       include: [/\.vue$/, /\.setup\.[cm]?[jt]sx?$/],
    }
  },
```

And:

```ts
import vueMacros from 'unplugin-vue-macros/vite'
import vue from '@vitejs/plugin-vue'

...
 vite: {
    vue: {
      script: {
        defineModel: true,
        propsDestructure: true,
      },
    }, 
    
    plugins: [
      vueMacros(),
      vue({
        include: [/\.vue$/, /\.setup\.[cm]?[jt]sx?$/],
        //                   ⬆️ setupSFC pattern need to be added
      }),
    ] as any,
  },
```

Also in the demo, there's no `"jsx": "preserve"`  in the tsconfig.json, but it's required to allow jsx in the project.

**Reply by angelhdzmultimedia** [reply]
Any idea what I could be possibly doing wrong? Thank you.

**Comment by angelhdzmultimedia** [comment]
### Solved?
(finally!?)

It's actually working. But there's a problem with Nuxt and VueRouter, that doesn't allow   .setup.tsx files to be in the same folder of the `~/pages`. I had `~/pages/index.vue` and `~/pages/test.setup.tsx`, and got the error in the image in my first comment.

After an exhausting amount of experimenting multime possible solutions, researching, 
finally decided to move the file to the ~/components directory and importing it from there as a component, and now it works!!!

But now this creates a new question, any way of telling VueRouter to use the setupSFC TSX syntax in its pages?
Nothing I add in `~/pages` or in any subfolder works.

**Comment by angelhdzmultimedia** [comment]
### Update

Yup, I'm afraid this is a bug. I'm going to move it to an Issue.
`.setup.tsx` files in pages should be parsed as well, and in Nuxt example there's not account for files inside the VueRouter `~/pages` directory.

**Comment by angelhdzmultimedia** [comment]
### Update 2

Yup. I removed `@vue-macros/nuxt` entirely, and tested Nuxt's native .tsx support for pages, and they work.

![image](https://github.com/vue-macros/vue-macros/assets/10191085/34fae46e-1429-4b90-b36b-f2742ca3e48d)

Then I add back `@vue-macros/nuxt` module, and:

![image](https://github.com/vue-macros/vue-macros/assets/10191085/321ff885-d032-4cb3-b5a6-a3ca87b072c8)

And yes, I have `macros: { setupSFC: true }` in `nuxt.config.ts`.

**Comment by angelhdzmultimedia** [comment]
### I give up 😔

My last attempt, getting rid of Nuxt module, and adding the Vite plugins instead:

![image](https://github.com/vue-macros/vue-macros/assets/10191085/ab3ef0a6-f4d8-4774-97fe-d7f5a31d8fdd)

And:

![image](https://github.com/vue-macros/vue-macros/assets/10191085/28b649fe-7ff7-45f4-a845-c5591cc0e12d)

And:

![image](https://github.com/vue-macros/vue-macros/assets/10191085/224c862a-cdf9-41a6-86c9-cdb5c78d8673)

**Reply by sxzz** [reply]
Could you please provide a reproduction repo?

**Reply by angelhdzmultimedia** [reply]
https://stackblitz.com/edit/nuxt-starter-zfbrrv

**Comment by sxzz** [comment, marked as answer]
Please try `@vue-macros/nuxt@1.6.0`. Thanks for your detailed feedback! ❤️ 
https://stackblitz.com/edit/nuxt-starter-ugvg9h?file=pages%2Findex.setup.tsx

**Reply by angelhdzmultimedia** [reply]
Ooohhh yes, they DO work NOW 😁, thanks to your fast, and hard work! 😍🤯⚡⏰

I appreciate it a lot. 

Time to have fun with setupSFC! 🔥🎉💚
